src/cli/proto_resolver.rs

```diff
@@ -0,0 +1,59 @@
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::Context;
+use protox::file::{FileResolver, GoogleFileResolver};
+use url::Url;
+
+use crate::{FileIO, HttpIO, ProtoPathResolver};
+
+#[derive(Clone)]
+pub struct NativeProtoPathResolver;
+
+impl NativeProtoPathResolver {
+    pub fn new() -> Self {
```

Comment:
Move http and file dependencies to `new` constructor. Make `resolve` method lighter.